test/1_stdlib/Unicode.swift

```diff
@@ -57,4 +57,12 @@ UnicodeAPIs.test("UnicodeDecodingResult/Equatable") {
   checkEquatable(instances, oracle: ==)
 }
 
+let UnicodeScalarTests = TestSuite("UnicodeScalar")
+
+UnicodeScalarTests.test("lossless conversion") {
+  let scalar: UnicodeScalar = "a"
+  let actual = UnicodeScalar(String(scalar))
+  expectOptionalEqual(scalar, actual)
```

Comment:
Needs a more robust test.  Perhaps the same lawlike forms for other conversion stuff.

```swift
forall us : UnicodeScalar, UnicodeScalar(String(us)) == us
```